### PR TITLE
New Feature - Edit Dimension Data Firewall Rule

### PR DESCRIPTION
**Pull request by samuelchong**
### Description

Implemented a new feature to DD libcloud to Edit Firewall Rule. This feature didn't exists before but it is now made available. Firewall rule will also accept portlist and ipaddresslist. Code snippet provided in the comment section of each method.
### Status
- done, ready for review
### Checklist (tick everything that applies)
- [Y] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [Y] Documentation
- [Y] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by tonybaloney** [comment]
thanks @samuelchong the change you submitted doesn't work in Python 2.6, also there are some issues with the linter.

https://travis-ci.org/apache/libcloud/builds/165730443

File: libcloud/test/compute/test_dimensiondata.py [hunk 1]
```diff
@@ -1185,6 +1382,370 @@ def test_detailed_usage_report(self):
...
+        child_ip_address_list = DimensionDataChildIpAddressList(
+            id=child_ip_address_list_id,
+            name="test_child_ip_addr_list")
+        net_domain = self.driver.ex_list_network_domains()[0]
+        ip_address_1 = DimensionDataIpAddress(begin='190.2.2.100')
+        ip_address_2 = DimensionDataIpAddress(begin='190.2.2.106',
+                                              end='190.2.2.108')
+        ip_address_3 = DimensionDataIpAddress(begin='190.2.2.0',
+                                              prefix_size='24')
+        ip_address_collection = {ip_address_1, ip_address_2,
```
Comment (tonybaloney) [inline libcloud/test/compute/test_dimensiondata.py]: you can't build sets like this in Python 2.6. Did you mean this to be a set?

File: libcloud/common/dimensiondata.py [hunk 2]
```diff
@@ -1510,3 +1520,233 @@ def __init__(self, id, name, description,
...
+        :type  network_adapter: ``str``
+        """
+        self.private_ip_v4 = private_ip_v4
+        self.vlan = vlan
+        self.network_adapter_name = network_adapter
+
+    def __repr__(self):
+        return ('<DimensionDataNic: private_ip_v4=%s, vlan=%s,'
+                'network_adapter=%s>'
+                % (self.private_ip_v4, self.vlan, self.network_adapter))
```
Comment (tonybaloney) [inline libcloud/common/dimensiondata.py]: this should be `self.network_adapter_name`

File: libcloud/test/compute/test_dimensiondata.py (hunk 1 above)
Comment (samuelchong) [reply libcloud/test/compute/test_dimensiondata.py]: Meant to use square bracket for a list. Syntax has been fixed and verified in python 2.7. Thx for your comment.

File: libcloud/common/dimensiondata.py (hunk 2 above)
Comment (samuelchong) [reply libcloud/common/dimensiondata.py]: Fixed to use self.network_adapter_name . Thx @tonybaloney

**Comment by tonybaloney** [comment]
thanks @samuelchong changes look good. 👍

**Comment by samuelchong** [comment]
hooray. edit FW is merged into trunk.

**Pull request merged**